### PR TITLE
Teacher Panel React Conversion: Stage Extras

### PR DESCRIPTION
**Pull request by dmcavoy**
Moves Stage Extras over to using the new react Teacher Panel to display student progress. Moves to using the same bubble display we use in the Progress Tab of Teacher Dashboard.

# Stage Extra Summary Level
[Example Level](http://localhost-studio.code.org:3000/s/csd3-2018/stage/3/extras?newTeacherPanel=1)

## Not Tried
| Before | After | 
| ------------- | ------------- | 
| Screen Shot 2019-05-23 at 11 08 12 AM | Screen Shot 2019-05-23 at 11 04 25 AM |

## Completed At Least One
| Before | After | 
| ------------- | ------------- | 
| Screen Shot 2019-05-23 at 11 06 44 AM | Screen Shot 2019-05-23 at 11 06 23 AM | 

# Stage Extra Sub Level
[Example Level](http://localhost-studio.code.org:3000/s/csd3-2018/stage/3/extras?id=23250&newTeacherPanel=1)

## Not Tried
| Before | After | 
| ------------- | ------------- | 
| Screen Shot 2019-05-23 at 11 18 36 AM | Screen Shot 2019-05-23 at 11 13 18 AM |

## Completed
| Before | After | 
| ------------- | ------------- | 
| Screen Shot 2019-05-23 at 11 08 45 AM | Screen Shot 2019-05-23 at 11 12 46 AM | 

## Completed Paired
| Before | After | 
| ------------- | ------------- | 
| Screen Shot 2019-05-23 at 11 18 05 AM | Screen Shot 2019-05-23 at 11 13 06 AM |

### CODE REVIEW

**Review by davidsbailey — APPROVED** [review]
This is looking great, Dani! more tests never hurt, but I think the test coverage on summarize_for_teacher_panel is looking good enough.

File: apps/src/sites/studio/pages/levels/_teacher_panel.js [hunk 1]
```diff
@@ -25,7 +25,10 @@ function initPage() {
   initViewAs(store);
   queryLockStatus(store, teacherPanelData.script_id);
   store.dispatch(getHiddenStages(teacherPanelData.script_name, false));
-  renderTeacherContentToggle(store);
+  // Stage Extras fail to load with this
+  if (!teacherPanelData.stage_extra) {
+    renderTeacherContentToggle(store);
```
Comment (davidsbailey) [inline apps/src/sites/studio/pages/levels/_teacher_panel.js:30]: this seems fine, we shouldn't need to ever toggle content on stage extras levels.

File: dashboard/app/models/script_level.rb [hunk 2]
```diff
@@ -372,6 +372,26 @@ def summarize_as_bonus
     }.camelize_keys
   end
 
+  def self.summarize_as_bonus_for_teacher_panel(script, bonus_level_ids, student)
```
Comment (davidsbailey) [inline dashboard/app/models/script_level.rb:375]: very nice!

**Review by davidsbailey — APPROVED** [review]
Looks great!

**Review by maddiedierker — APPROVED** [review]
+1 to dave's comments -- looking great, dani!

**Pull request merged**